### PR TITLE
Rename package to nisystemlink

### PR DESCRIPTION
**Pull request by cameronwaterman**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nisystemlink-clients-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Renames the package to "nisystemlink" since the existing name "systemlink" conflicts with the SystemLink SDK which would prevent it from being used in Jupyter.

### Why should this Pull Request be merged?

Provides a unique package name so it can be used in Jupyter.

### What testing has been done?

- Ran tests
- Ran linter
- Built package and inspected ouput

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -43,9 +43,10 @@ poethepoet               = "^0.16.4"
 [tool.poe.tasks]
 test       = "pytest tests -m \"(not slow) and (not cloud)\""
 doctest    = "pytest --doctest-modules docs"
-format     = "black --exclude \".*\\.pyi\" --check systemlink examples tests"
-lint       = "flake8 systemlink examples tests"
-types      = "mypy --config-file mypy.ini systemlink examples/tag tests"
+check      = "black --exclude \".*\\.pyi\" --check nisystemlink examples tests"
```
Comment (cameronwaterman) [inline pyproject.toml:46]: @mure - I changed "format" to "check" and updated the workflow to call this one instead, and then added a new "format" task that actually performs the format.

Comment (mure) [reply pyproject.toml:46]: Makes sense - thanks

**Review by mure — APPROVED** [review]

File: nisystemlink/clients/core/_internal/_winpaths.pyi [hunk 2]
```diff
@@ -4,11 +4,9 @@ import ctypes
 from ctypes import wintypes
 from uuid import UUID
 
-
```
Comment (spanglerco) [inline nisystemlink/clients/core/_internal/_winpaths.pyi:7]: I'm curious about tehse edits come from, and for stubs/events.pyi as well. Aren't there usually two blank lines between classes? Also, it looks as though the black command is intended to exclude pyi files.

**Review by spanglerco — APPROVED** [review]

File: nisystemlink/clients/core/_internal/_winpaths.pyi (hunk 2 above)
Comment (cameronwaterman) [reply nisystemlink/clients/core/_internal/_winpaths.pyi:7]: These changes did come from black, (running `poe format`) - so it looks like there might be an issue with the regex that is supposed to exclude pyi files

Comment (cameronwaterman) [reply nisystemlink/clients/core/_internal/_winpaths.pyi]: Looks like there was a Windows-specific issue when specifying the `--exclude` regex inline in the poe task. Moving the exclude config out to a black-specific section in the pyproject.toml file seems to have fixed it.

**Pull request merged**